sechub-commons-model/src/main/java/com/mercedesbenz/sechub/commons/model/SecHubRemoteCredentialSupport.java

```diff
@@ -0,0 +1,65 @@
+package com.mercedesbenz.sechub.commons.model;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SecHubRemoteCredentialSupport {
```

Comment:
I thought about the pattern compile problem (needs time).

I have following idea:

divide the logic:

CreentialDataContainer container = support -> createCredentialDataContainer(SecHubRemoteCredentialConfiguration configuration)
( can be created on startup of a PDS instance for example)

the create method will compiles all patterns inside and create a "container" which
has the patterns inside at runtime.

at runtime the container is used by logic and provides the resolve methods.

Remark: I speak for container this time - means after the renaming. we would have here
- `SecHubRemoteCredentialConfiguration` contains json structure with `SecHubRemoteCredentialData` inside
- `SecHubRemoteCredentialData` (data)
- `SecHubRemoteCredentialSupport` (support which can create the container) - but we could rename it to `SecHubRemoteCredentialDataFactory` as well, becauseit only creates the container...
- `SecHubRemoteCredentialContainer` (resolve methods)